### PR TITLE
1.12.0

### PR DESCRIPTION
**Pull request by lucasborin**

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -12,6 +12,10 @@ Upgrade Note
 ------------------
 Whenever you upgrade code pal for ABAP, it is highly recommended to execute the Y_CI_CHECK_REGISTRATION report to activate/reactivate the Checks (SE38 transaction) and regenerate the respective code inspector variant (SCI transaction)
 
+2021-01-XX v1.12.0
+------------------
+- RFC Enabled (#305)
```
Comment (larshp) [inline changelog.txt:17]: disabled?

Comment (lucasborin) [reply changelog.txt:17]: Fixed. Thanks for your review. :)

File: src/foundation/y_check_base.clas.abap [hunk 2]
```diff
@@ -169,16 +161,11 @@ CLASS y_check_base IMPLEMENTATION.
     IF ref_scan_manager->is_scan_ok( ) = abap_false.
       RAISE EXCEPTION TYPE ycx_object_not_processed.
     ENDIF.
-
-    IF clean_code_exemption_handler->is_object_exempted( object_name = object_name object_type = object_type ) = abap_true.
-      RAISE EXCEPTION TYPE ycx_object_is_exempted.
```
Comment (lucasborin) [inline src/foundation/y_check_base.clas.abap:174]: as it is not being raised here anymore, we have to remove its catch on the caller side

File: src/foundation/y_check_base.clas.abap [hunk 3]
```diff
@@ -636,13 +628,29 @@ CLASS y_check_base IMPLEMENTATION.
...
+
+    TRY.
+        DATA(main_app_comp) = y_code_pal_app_comp=>get( ref_scan_manager->levels[ level = 0 ] ).
+        DATA(curr_app_comp) = y_code_pal_app_comp=>get( ref_scan_manager->levels[ statement_level ] ).
+
+        IF main_app_comp <> curr_app_comp.
+          RETURN.
+        ENDIF.
+      CATCH cx_sy_itab_line_not_found
+            ycx_entry_not_found.
```
Comment (lucasborin) [inline src/foundation/y_check_base.clas.abap:643]: empty catch. Move the logic to a method that returns a boolean.

File: src/foundation/y_exemption_handler.clas.abap [hunk 4]
```diff
@@ -42,16 +43,30 @@ CLASS y_exemption_handler IMPLEMENTATION.
...
+                       WHEN object_type = 'FUGR' THEN y_exemption_of_function_group=>is_exempted( object_name ) ).
+    ENDIF.
+
+    APPEND VALUE #( object_type = object_type
+                    object_name = object_name
+                    is_exempted = result ) TO buffer.
+
+    IF lines( buffer ) > max_entries.
+      DELETE buffer FROM 1 TO max_entries / 2.
+    ENDIF.
```
Comment (lucasborin) [inline src/foundation/y_exemption_handler.clas.abap:69]: Run it first as done it appl_comp.

File: src/foundation/y_object_creation_date.clas.abap [hunk 5]
```diff
@@ -1,81 +1,115 @@
...
+
+    result = get_lowest_date( created_on_dates  ).
+
+    APPEND VALUE #( object_type = object_type
+                    object_name = object_name
+                    created_on = result ) TO buffer.
+
+    IF lines( buffer ) > max_entries.
+      DELETE buffer FROM 1 TO max_entries / 2.
+    ENDIF.
```
Comment (lucasborin) [inline src/foundation/y_object_creation_date.clas.abap:109]: Move to the beginning of the method.

**Pull request merged**